File: specs/query_dsl/compound/function_score/function_score_query.ts

/**namespace:QueryDsl.Compound.FunctionScore */
/**custom_serialization*/
interface function_score_query {
	query: query_container;
	functions: score_function[];
	max_boost: double;
	score_mode: FunctionScoreMode;
	boost_mode: FunctionBoostMode;
	min_score: double;
}

File: output/csharp/QueryDsl/Compound/FunctionScore/FunctionScoreQuery.cs
using Nest.QueryDsl;
using Nest.Internal;
using System;
using System.Collections.Generic;
using System.Runtime.Serialization;
namespace Nest.QueryDsl {

	public class FunctionScoreQuery  {
		
		[DataMember(Name="boost_mode")]
		public FunctionBoostMode BoostMode { get; set; }


		[DataMember(Name="functions")]
		public List<ScoreFunction> Functions { get; set; }


		[DataMember(Name="max_boost")]
		public double MaxBoost { get; set; }


		[DataMember(Name="min_score")]
		public double MinScore { get; set; }


		[DataMember(Name="query")]
		public QueryContainer Query { get; set; }


		[DataMember(Name="score_mode")]
		public FunctionScoreMode ScoreMode { get; set; }

	}
}
